pkg/proxy/healthcheck/healthcheck.go

```diff
@@ -211,7 +211,7 @@ func (h hcHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 
 	resp.Header().Set("Content-Type", "application/json")
 	if count == 0 {
-		resp.WriteHeader(http.StatusServiceUnavailable)
+		http.Error(resp, "", http.StatusServiceUnavailable)
```

Comment:
this resets the content type and makes the following json an invalid response, right?

Reply:
> localEndpoints will be equal to 0 in case of an error. Is that the right behavior?

that would be a good question for @kubernetes/sig-network-api-reviews, but my leaning would be toward keeping the current behavior

Reply:
I've updated the PR to keep the current behavior here.